libp2p/NodeTable.cpp

```diff
@@ -320,22 +326,30 @@ vector<shared_ptr<NodeEntry>> NodeTable::nearestNodeEntries(NodeID _target)
 
 void NodeTable::ping(NodeEntry const& _nodeEntry, boost::optional<NodeID> const& _replacementNodeID)
 {
-    m_timers.schedule(0, [this, _nodeEntry, _replacementNodeID](
-                             boost::system::error_code const& _ec) {
+    // don't sent Ping if one is already sent
+    if (contains(m_sentPings, _nodeEntry.id))
+        return;
+
+    NodeIPEndpoint src;
+    src = m_hostNodeEndpoint;
+    PingNode p(src, _nodeEntry.endpoint);
+    p.ts = nextRequestExpirationTime();
+    auto const pingHash = p.sign(m_secret);
+    LOG(m_logger) << p.typeName() << " to " << _nodeEntry.id << "@" << p.destination;
```

Comment:
Can we just log the _nodeEntry here to get both the ID and the destination endpoint?